### PR TITLE
Dịch single-variable-calculus_vn.md phần 1.

### LINKED ISSUE #389: Dịch "chapter_appendix_math/single-variable-calculus_vn" - Phần 01

### PR DESCRIPTION
**Pull request by lkhphuc**
## Pull Request checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #389 

* [x] Bản dịch này có bám sát [Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md) không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm
    * Tiêu đề (số lượng dấu `#` đầu dòng)
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`)
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt)
    * Không dịch các phần code (nằm giữa hai dấu `````)
    * Copy các công thức toán từ bản gốc (các đoạn có `$`)
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`)
    * Không tự thêm bớt các dòng trắng
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm [hướng dẫn hoặc yêu cầu dịch hình](../img/README.md).

* [x] Điền tên của bạn và vào mục "Những người thực hiện" ở cuối file.

* [x] Pull Request này đã sẵn sàng cho việc Review chưa?

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này.

## Với reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 1]
```diff
@@ -4,47 +4,52 @@
...
+Đây là một hàm số cực kì phức tạp, chứa đựng chất lượng của tất cả mô hình có thể của một cấu trúc cho trước trên tập dữ liệu này, do vậy hầu như là không thể chỉ ra được một tập hợp các trọng số $\mathbf{w}$ nào sẽ tối thiểu hoá mất mát.
+Do vậy trên thực tế, chúng ta thường bắt đầu bằng việc khởi tạo *ngẫu nhiên* các trọng số, và đi từng bước nhỏ theo hướng mà sẽ giảm giá trị mất mát nhanh nhất có thể.
 
 <!--
 The question then becomes something that on the surface is no easier: how do we find the direction which makes the weights decrease as quickly as possible?  To dig into this, let's first examine the case with only a single weight: $L(\mathbf{w}) = L(x)$ for a single real value $x$.
 -->
 
-*dịch đoạn phía trên*
+Vấn đề trên bây giờ trở thành một câu hỏi mà thoạt nhìn cũng không dễ hơn bao nhiêu: làm thế nào để tìm được hướng đi sẽ giảm giá trị mất mát nhanh nhất có thể? <!-- Cái này bản gốc viết sai rồi, sao lại giảm weights nhanh nhất có thể -->
+Chúng ta sẽ đi sâu vào câu hỏi trên trong trường hợp chỉ có một trọng số: $L(\mathbf{w}) = L(x)$  cho một số thực $x$ duy nhất.
```
Comment (rootonchair) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Để trả lời câu hỏi này, trước hết ta hãy xét trường hợp chỉ có một trọng số: $L(\mathbf{w}) = L(x)$  cho một số thực $x$ duy nhất.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 2]
```diff
@@ -4,47 +4,52 @@
...
-*dịch đoạn phía trên*
+Trong :numref:`sec_calculus`,  chúng ta đã xem qua những thành phần cơ bản nhất của giải tích vi phân.
+Trong mục này chúng ta sẽ đi sâu vào kiến thức giải tích cơ bản và cách áp dụng chúng trong trong ngữ cảnh học máy.
 
 <!--
 ## Differential Calculus
 -->
 
-## *dịch tiêu đề phía trên*
+## Vi phân
```
Comment (rootonchair) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
## Giải tích vi phân
```

Mình nghĩ là nên ghi đầy đủ

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 3]
```diff
@@ -4,47 +4,52 @@
...
 
 <!--
 Differential calculus is fundamentally the study of how functions behave under small changes.  To see why this is so core to deep learning, let's consider an example.
 -->
 
-*dịch đoạn phía trên*
+Vi phân là nhánh toán học nghiên cứu về hành vi của các hàm số dưới các biến đổi nhỏ.
```
Comment (rootonchair) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Giải tích vi phân là nhánh toán học nghiên cứu về hành vi của các hàm số dưới các biến đổi nhỏ.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 4]
```diff
@@ -788,7 +793,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 -->
 
 <!-- Phần 1 -->
-*
+* Lê Khắc Hồng Phúc
```
Comment (lkhphuc) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
* Lê Khắc Hồng Phúc
* Phạm Hồng Vĩnh
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 5]
```diff
@@ -4,47 +4,52 @@
 # Single Variable Calculus
 -->
 
-# *dịch tiêu đề phía trên*
+# Giải tích đơn biến
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]: Mình nhớ có nhiều sách dùng "giải tích một biến" và "giải tích nhiều biến".

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 6]
```diff
@@ -4,47 +4,52 @@
...
 :label:`sec_single_variable_calculus`
 
 <!--
 In :numref:`sec_calculus`, we saw the basic elements of differential calculus. 
 This section takes a deeper dive into the fundamentals of calculus and how we can understand and apply it in the context of machine learning.
 -->
 
-*dịch đoạn phía trên*
+Trong :numref:`sec_calculus`,  chúng ta đã xem qua những thành phần cơ bản nhất của giải tích vi phân.
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Trong :numref:`sec_calculus`, chúng ta đã thấy những thành phần cơ bản của giải tích vi phân.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 7]
```diff
@@ -4,47 +4,52 @@
...
+Trong mục này chúng ta sẽ đi sâu vào kiến thức giải tích cơ bản và cách áp dụng chúng trong trong ngữ cảnh học máy.
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Trong mục này chúng ta sẽ đi sâu vào kiến thức nền tảng của giải tích và cách áp dụng chúng trong trong ngữ cảnh học máy.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 8]
```diff
@@ -4,47 +4,52 @@
...
+Để thấy được tại sao đây lại là cốt lõi của học sâu, hãy cùng xem ví dụ dưới đây.
 
 <!--
 Suppose that we have a deep neural network where the weights are, for convenience, concatenated into a single vector $\mathbf{w} = (w_1, \ldots, w_n)$.  
 Given a training dataset, we consider the loss of our neural network on this dataset, which we will write as $\mathcal{L}(\mathbf{w})$.
 -->
 
-*dịch đoạn phía trên*
+Giả sử chúng ta có một mạng nơ-ron sâu với các trọng số được biễu diễn bằng một vector duy nhất $\mathbf{w} = (w_1, \ldots, w_n)$.
+Cho trước một tập huấn luyện, chúng ta sẽ tập trung vào giá trị mất mát của mạng nơ-ron trên tập huấn luyện đấy và kí hiệu nó bằng $\mathcal{L}(\mathbf{w})$.
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Cho trước một tập huấn luyện, chúng ta sẽ tập trung vào giá trị mất mát của mạng nơ-ron trên tập huấn luyện đấy và ký hiệu nó bằng $\mathcal{L}(\mathbf{w})$.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 9]
```diff
@@ -4,47 +4,52 @@
...
 
 <!--
 This function is extraordinarily complex, encoding the performance of all possible models of the given architecture on this dataset, so it is nearly impossible to tell what set of weights $\mathbf{w}$ will minimize the loss. 
 Thus, in practice, we often start by initializing our weights *randomly*, and then iteratively take small steps in the direction which makes the loss decrease as rapidly as possible.
 -->
 
-*dịch đoạn phía trên*
+Đây là một hàm số cực kì phức tạp, chứa đựng chất lượng của tất cả mô hình có thể của một cấu trúc cho trước trên tập dữ liệu này, do vậy hầu như là không thể chỉ ra được một tập hợp các trọng số $\mathbf{w}$ nào sẽ tối thiểu hoá mất mát.
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Đây là một hàm số cực kì phức tạp, là dạng tổng quát của tất cả mô hình có thể của một cấu trúc cho trước trên tập dữ liệu này, do vậy gần như không thể chỉ ra được một tập hợp các trọng số $\mathbf{w}$ để tối thiểu hoá mất mát.
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 10]
```diff
@@ -4,47 +4,52 @@
...
+Do vậy trên thực tế, chúng ta thường bắt đầu bằng việc khởi tạo *ngẫu nhiên* các trọng số, và đi từng bước nhỏ theo hướng mà sẽ giảm giá trị mất mát nhanh nhất có thể.
 
 <!--
 The question then becomes something that on the surface is no easier: how do we find the direction which makes the weights decrease as quickly as possible?  To dig into this, let's first examine the case with only a single weight: $L(\mathbf{w}) = L(x)$ for a single real value $x$.
 -->
 
-*dịch đoạn phía trên*
+Vấn đề trên bây giờ trở thành một câu hỏi mà thoạt nhìn cũng không dễ hơn bao nhiêu: làm thế nào để tìm được hướng đi sẽ giảm giá trị mất mát nhanh nhất có thể? <!-- Cái này bản gốc viết sai rồi, sao lại giảm weights nhanh nhất có thể -->
```
Comment (tiepvupsu) [inline chapter_appendix_math/single-variable-calculus_vn.md]: Đúng rồi, bản gốc bị sai.
```suggestion
Vấn đề trên bây giờ trở thành một câu hỏi mà thoạt nhìn cũng không dễ hơn bao nhiêu: làm thế nào để tìm được hướng đi sẽ giảm giá trị hàm mất mát nhanh nhất có thể? <!-- Cái này bản gốc viết sai rồi, sao lại giảm weights nhanh nhất có thể -->
```

File: chapter_appendix_math/single-variable-calculus_vn.md (hunk 9 above)
Comment (rootonchair) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
Đây là một hàm số cực kỳ phức tạp, chứa đựng chất lượng của tất cả mô hình có thể của một cấu trúc cho trước trên tập dữ liệu này, do vậy hầu như là không thể chỉ ra được một tập hợp các trọng số $\mathbf{w}$ nào sẽ tối thiểu hoá mất mát.
```

File: chapter_appendix_math/single-variable-calculus_vn.md (hunk 5 above)
Comment (lkhphuc) [reply chapter_appendix_math/single-variable-calculus_vn.md]: Em googgle thử thì thấy toàn xài `một biến` `nhiều biến` thật.
```suggestion
# Giải tích một biến
```

File: chapter_appendix_math/single-variable-calculus_vn.md [hunk 11]
```diff
@@ -788,7 +793,8 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 -->
 
 <!-- Phần 1 -->
-*
+* Lê Khắc Hồng Phúc
+* Phạm Hồng Vĩnh
```
Comment (lkhphuc) [inline chapter_appendix_math/single-variable-calculus_vn.md]:
```suggestion
* Phạm Hồng Vĩnh
* Vũ Hữu Tiệp
```
